File: src/api/ai.ts
import { apiJson, apiRequest } from "./client";

export type AIAnalyzeRequest = {
  symbol: string;
  task: string;
  provider?: string;
  model?: string;
  context?: Record<string, unknown>;
};

export type AIAnalysis = {
  id: string;
  workflow_id?: string | null;
  provider: string;
  model: string;
  task: string;
  summary: string;
  signals: string[];
  risks: string[];
  recommendation: string;
  confidence: number;
  created_at: string;
  is_mock: boolean;
  source: string;
  status: string;
};

export function analyzeWithBackendAI(payload: AIAnalyzeRequest) {
  return apiJson<AIAnalysis>("/api/ai/analyze", {
    ...payload,
  });
}

export type AIProviderStatus = {
  name: string;
  configured: boolean;
  healthy: boolean | null;
  default_model: string;
  base_url: string;
  capabilities: {
    streaming: boolean;
    json_mode: boolean;
    tools: boolean;
    vision: boolean;
    reasoning: boolean;
  };
  last_error?: string | null;
};

export type AIProviderTestResult = {
  ok: boolean;
  provider: string;
  model: string;
  latency_ms: number;
  error_type?: string | null;
  error_message?: string | null;
};

export function fetchAIProviders(signal?: AbortSignal) {
  return apiRequest<AIProviderStatus[]>("/api/ai/providers", { signal });
}

export function testAIProvider(provider: string, model?: string) {
  return apiJson<AIProviderTestResult>("/api/ai/providers/test", { provider, model });
}

File: backend/app/schemas/ai.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.common import SourceMeta


class AIAnalyzeRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    symbol: str = "ETH/USDT"
    task: str = "market_analysis"
    provider: str | None = None
    model: str | None = None
    context: dict[str, Any] = Field(default_factory=dict)


class AIAnalysisRead(SourceMeta):
    id: str
    workflow_id: str | None = None
    provider: str
    model: str
    task: str
    summary: str
    signals: list[str]
    risks: list[str]
    recommendation: str
    confidence: int
    ai_analysis_id: str | None = None
    created_at: datetime


class AIAnalysisResponse(BaseModel):
    ok: Literal[True] = True
    data: AIAnalysisRead


class AIProviderTestRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    provider: str
    model: str | None = None


class AIProviderTestRead(BaseModel):
    ok: bool
    provider: str
    model: str
    latency_ms: int
    error_type: str | None = None
    error_message: str | None = None


class AIProviderModelsRead(BaseModel):
    provider: str
    configured: bool
    models: list[str]
    default_model: str
